plugins/event_source/cmd.py

```diff
@@ -0,0 +1,107 @@
+"""
+cmd.py
+
+An ansible-rulebook event source plugin for running a custom command.
+It can send events to the rule engine from the stdout.
+
+Arguments:
+    command: str
+        The string with the command to execute
+    send_output: bool, default: false
+        Send events from the stdout.
+        Each event must be a serialized JSON string
+        ended with a newline character ('\n')
+
+Examples:
+---
+- name: Example with output proccessing
+  hosts: all
+  sources:
+    - ansible.eda.cmd:
+        command: node my_custom_plugin.js
+        send_output: true
+  rules:
+    - name: r1
+      condition: event.cmd.status == "Running"
+      action:
+        print_event:
+
+
+---
+- name: Example combined with webhook plugin
+  hosts: all
+  sources:
+    - ansible.eda.webhook:
+        port: 6080
+    - ansible.eda.cmd:
+        command: node my_custom_plugin.js --url http://localhost:6080/eda
+  rules:
+    - name: r1
+      condition: event.payload.status == "Running"
+      action:
+        print_event:
+"""
+
+import asyncio
+import json
+import logging
+import shlex
+from typing import Any, Dict
+
+LOGGER = logging.getLogger("ansible.eda.cmd")
+
+
+async def process_output(
+    proc: asyncio.subprocess.Process, queue: asyncio.Queue
+):
+    # type hint warn: proc.stdout can be None, this is not our case
+    while output := await proc.stdout.readline():
+        try:
+            event = json.loads(output.decode())
+            await queue.put({"cmd": event})
+        except json.JSONDecodeError as e:
+            LOGGER.error("Can not deserialize JSON data: %s", e)
```

Comment:
Consider simply `put({"cmd": output.decode()})`. The user can then use a customized filter to analyze the string and convert it to a proper json body.

Reply:
I wanted to leave that constraint in the cmd, but I think give the option is not a bad idea. I will add a new option for it